node/src/consensus/consensus_worker.rs

```diff
@@ -0,0 +1,107 @@
+use super::{communications::*, errors::ConsensusError, ConsensusCommittee};
+use crate::{
+    config::NodeConfig,
+    consensus::LOG_TARGET,
+    db::utils::db::db_client,
+    types::{consensus::CommitteeState, NodeID},
+};
+use log::{error, warn};
+
+pub struct ConsensusWorker {
+    node_config: NodeConfig,
+}
+
+impl ConsensusWorker {
+    pub fn new(node_config: NodeConfig) -> Result<Self, ConsensusError> {
+        Ok(ConsensusWorker { node_config })
+    }
+
+    pub fn work(&self, node_id: NodeID) -> Result<(), ConsensusError> {
+        let config = self.node_config.clone();
+        actix_rt::spawn(async move {
+            if let Err(e) = ConsensusWorker::task(node_id, config).await {
+                error!("ConsensusWorker work error: {}", e)
+            };
+        });
+
+        Ok(())
+    }
+
+    async fn task(node_id: NodeID, node_config: NodeConfig) -> Result<bool, ConsensusError> {
+        let client = db_client(&node_config)
+            .await
+            .expect("Validator node unable to load db client");
```

Comment:
This is good for current stage IMHO. I think we may want to amend this in the future to more actor-like approach, based on pub-sub to provide faster feedback, also that will stay in separate thread pool in the future